iroh/src/magicsock/transports.rs

```diff
@@ -34,18 +34,19 @@ pub(crate) struct Transports {
     ip: Vec<IpTransport>,
     relay: Vec<RelayTransport>,
 
-    poll_recv_counter: AtomicUsize,
+    poll_recv_counter: usize,
+    source_addrs: tinyvec::TinyVec<[Addr; 4]>, // cache for source addrs
```

Comment:
4 might be a little conservative, depending on if your system has GSO enabled or not. E.g. I consistently get `32` as the value for `metas.len()`.

Reply:
I don't want to fill the stack that much, we already fill it quite a bit. So I would only want to increase this if we have benchmarks showing this is clearly worth it

Reply:
32 * `Addr` = 1280 bytes. The `Transport` struct is put on the stack in the `async fn Handle::new`. Those cases *can* cause ballooning stack sizes when the struct has to be held across an await point, which currently isn't the case.
In fact, it's moved into `MagicTransport::new` which immediately gets `Box::new`d.

I don't think that 1.3kB on the stack is too bad.

But also, given it's moved into a `Box`, doesn't that mean it'll live on the heap anyways?
Is there a performance difference to using a `Vec`?

Reply:
> Is there a performance difference to using a `Vec`?

yeah, at least in my tests